WinImageBuilder.psm1

```diff
@@ -670,6 +670,46 @@ function Compress-Image {
     Write-Output "MaaS image is ready and available at: $ImagePath"
 }
 
+function Start-Executable {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)]
+        [Alias("7za.exe")]
+        [array]$Command
+    )
+    PROCESS {
+        $cmdType = (Get-Command $Command[0]).CommandType
+        if ($cmdType -eq "Application") {
+            $ErrorActionPreference = "SilentlyContinue"
+            $ret = & $Command[0] $Command[1..$Command.Length] 2>&1
+            $ErrorActionPreference = "Stop"
+        } else {
+            $ret = & $Command[0] $Command[1..$Command.Length]
+        }
+        if ($cmdType -eq "Application" -and $LASTEXITCODE) {
+            Throw ("Failed to run: " + ($Command -Join " "))
+        }
+        if ($ret -and $ret.Length -gt 0) {
+            return $ret
+        }
+        return $false
+    }
+}
+
+function New-ProtectedZip {
+    Param(
+        [parameter(Mandatory=$true)]
+        [string]$ZipPassword,
+        [Parameter(Mandatory=$true)]
+        [string]$VirtualDiskPath
+    )
+        $zipPath = (Get-PathWithoutExtension $VirtualDiskPath) + ".zip"
+        $7zip = Join-Path $localResourcesDir 7za.exe

```

Comment:
Create a Get-7zipPath method, which returns the binary path. Refactorhttps://github.com/cloudbase/windows-openstack-imaging-tools/blob/master/WinImageBuilder.psm1#L627 method to also use it